doc/l2mc/IGMP_Snooping_HLD.md

```diff
@@ -0,0 +1,701 @@
+ 8. Support trapping IGMP and PIM Hello control packets to CPU.
+ 9. Support Link-layer topology changes (due to STP) and generate queries on all non-mrouter ports for faster convergence. 
+ 10. Support IS-CLI Configuration and show commands
+ 11. Support IGMP Snooping for MCLAG
+ 
+
+### 1.1.2 Configuration and Management Requirements
+
+To support IS-CLI configuration and show commands.
+ 1. Support CLI configurations as described in "Configuration Commands" section below
+ 2. Support show commands as described in "Show Commands" section below
+ 3. Support openconfig REST API and gNMI
+
+### 1.1.3 Scalability Requirements
+
+SONiC IGMP Snooping feature does not have scale limit, scale number varies based on ASIC supported number, refer to scalability section below for the tested scale numbers.
+ 
+### 1.1.4 Warm Boot Requirements
+
+Dynamic and static L2MC entries should persist across warm reboot with no traffic disruption to the active flows.
+And, remote learnt L2MC entries in MCLAG should persist across warm reboot.
+Learning of new snooping entries and aging of existing L2MC entries will stop while the control plane is away.
+- To support planned system warm boot.
+
+## 1.2 Design Overview
+
+### 1.2.1 Basic Approach
+### 1.2.2  L2MC Docker
+A new L2MC docker will be created to support IGMP snooping.
+L2MC will host l2mcastMgr and l2mcastd,  and l2mcastSync will be part of l2mcastd, the details of the changes will be discussed in the design section below.  
+
+### 1.2.3 IGMP control packets flooding 
+IGMP control packets flooding happens in kernel, kernel  flooding behaviour is per bridge, there is no per VLAN flooding control in Kernel.  Until the dynamic multicast-router(mrouter) port learns in the VLAN, IGMP reports will be flooded to all members of the VLAN.  
+In presence of multicast-router, IGMP reports received on a host interface are forwarded to multicast-router interfaces alone, but not to the other host interfaces in the VLAN.
+
+### 1.2.4 SAI Overview
+The existing L2MC SAI interface APIs shall be used to support this feature and they are available at below location,  
+[https://github.com/opencomputeproject/SAI/blob/master/inc/sail2mc.h](https://github.com/opencomputeproject/SAI/blob/master/inc/sail2mc.h)  
+https://github.com/opencomputeproject/SAI/blob/master/inc/sail2mcgroup.h
+
+The detail of the changes will be discussed in the design section below.
+
+### 1.2.5 MCLAG Overview
+In an MCLAG Enabled node, IGMP Source can be at MCLAG member port or Orphan Port. Details of IGMP changes will be discussed at design section below.
+
+# 2 Functionality
+Refer to section 1
+## 2.1 Target Deployment Use Cases
+
+## 2.2 Functional Description
+Refer to section 1.1
+
+# 3 Design
+## 3.1 Overview
+
+
+
+![IGMP Snooping](images/L2mc_Architecture_diagram.png "Figure 1: IGMP Snooping High Level Architecture")
+
```

Comment:
Diagrams are not accessible.. can you please check?

Reply:
I will check and update.